src/snowflake/snowpark/session.py

```diff
@@ -2284,4 +2284,22 @@ def _get_client_side_session_parameter(self, name: str, default_value: Any) -> A
             else default_value
         )
 
+    def __repr__(self):
+        """
+        Return a string representation of the Session object."""
+        snowflake_environment = self.sql("select current_version()").collect()
+        return (
```

Comment:
I would suggest using internal `_run_query` which returns a list. it will be lighter weight